### PR TITLE
Formatter now respects map assoc => and match := and adds spaces around

### LINKED ISSUE #839: reformat code with map
````erlang
#{a=>1}
%to
#{a => 1}
````

### PR DESCRIPTION
**Pull request by kvakvs**
Closes #839 
Formatter adds spaces around `=>` (`ERL_ASSOC`) and `:=` (`ERL_MATCH`) following option `SPACE_AROUND_ARROW`

### CODE REVIEW

**Comment by ignatov** [comment]
Please add a test, see https://github.com/ignatov/intellij-erlang/blob/master/tests/org/intellij/erlang/formatting/ErlangFormattingTest.java

**Comment by kvakvs** [comment]
Added a test.

File: tests/org/intellij/erlang/formatting/ErlangFormattingTest.java [hunk 1]
```diff
@@ -349,6 +349,12 @@ public void testUniformBinaryExpressionsStyle() throws Exception {
   public void testMapTupleAfterFirstEntry()  throws Exception { doEnterTest(); }
   public void testMapTupleAfterLastEntry()   throws Exception { doEnterTest(); }
 
+  public void testMapAssocSpacing() throws Exception {
+    ErlangCodeStyleSettings erlangSettings = getErlangSettings();
+    erlangSettings.SPACE_AROUND_ARROW = true;
```
Comment (ignatov) [inline tests/org/intellij/erlang/formatting/ErlangFormattingTest.java]: It's better to restore settings back.
```java
prev = erlangSettings.SPACE_AROUND_ARROW;
try {
  erlangSettings.SPACE_AROUND_ARROW = true;
  doTest();
} finally {
  erlangSettings.SPACE_AROUND_ARROW = prev;
}

Comment (kvakvs) [reply tests/org/intellij/erlang/formatting/ErlangFormattingTest.java]: Added

**Comment by kvakvs** [comment]
A configuration problem: 
On Windows tests pass or mostly pass
On Linux running tests fails every test with 
```
[0]: Unable to create a temporary file /tmp/unitTest_afterIf1/unitTest_337/unitTest_noReformatAfterInsertedText/unitTest_functionClause/unitTest_afterIf2/unitTest_ifBeforeIf/unitTest_afterExpression/unitTest_bindTryParasite__IR_OFF/unitTest_bindIfLastClause/unitTest_bindIfFirstClause__IR_OFF/unitTest_bindBeginExpr/unitTest_bindCase__IR_OFF/unitTest_bindBeginExpr__IR_OFF/unitTest_bindReceiveAfterClause__IR_OFF/unitTest_mapTupleBeforeFirstEntry/unitTest_bindCaseOfParasite__IR_OFF/unitTest_bindTryExprParasite/unitTest_bindReceiveAfterParasite/unitTest_bindTryCatch/unitTest_bindIfParasite__IR_OFF/unitTest_bindCaseOfFirstClauseParasite__IR_OFF/unitTest_recordExpressionAfterLastFieldAssignment/unitTest_bindTryCatch__IR_OFF/unitTest_bindTryFirstCatchClause__IR_OFF/unitTest_bindReceiveLastClause/unitTest_listAlignOnEnter/unitTest_binaryComprehensionAfterExpression/unitTest_commaFirstEnter2/unitTest_listComprehensionAfterOrOr/unitTest_funArgsBeforeFirst/unitTest_bindReceiveFirstClause/unitTest_116/unitTest_118/unitTest_125/unitTest_136/unitTest_137/unitTest_141/unitTest_171/unitTest_191/unitTest_222/unitTest_273/unitTest_288/unitTest_292/unitTest_299/unitTest_305/unitTest_350/unitTest_351/unitTest_379/unitTest_433/unitTest_434/unitTest_444/unitTest_451/unitTest_478/unitTest_754/unitTest_if1/unitTest_if2/unitTest_if3/unitTest_funArgsAfterFirst/unitTest_funCallAfterComma/unitTest_aligned/unitTest_case1/unitTest_case2/unitTest_case3/unitTest_case4/unitTest_recordExpressionBeforeFieldAssignment/unitTest_bindBegin__IR_OFF/unitTest_try1/unitTest_try2/unitTest_try3/unitTest_try4/unitTest_try5/unitTest_mapTupleAfterFirstEntry/unitTest_bindBeginParasite__IR_OFF/unitTest_bindCaseExpr__IR_OFF/unitTest_bindReceive__IR_OFF/unitTest_bindReceiveAfterClauseParasite__IR_OFF/unitTest_binaryComprehensionBeforeExpression/unitTest_functionClausesAligned/unitTest_funCallNoArgs/unitTest_binaryComprehensionAfterLastExpr/unitTest_bindIfLastClauseParasite/unitTest_bindTryFirstCatchClauseParasite__IR_OFF/unitTest_caseParasite1/unitTest_caseParasite2/unitTest_caseParasite3/unitTest_caseParasite4/unitTest_bindCaseOfFirstClause/unitTest_recordExpressionAfterFirstFieldAssignment/unitTest_bindCaseOf__IR_OFF/unitTest_48/unitTest_52/unitTest_53/unitTest_67/unitTest_71/unitTest_75/unitTest_82/unitTest_95/unitTest_bindCaseOfLastClause/unitTest_mapComprehensionBeforeExpression/unitTest_bindFunExpressionLastClause__IR_OFF/unitTest_bindBeginExprParasite__IR_OFF/unitTest_bindIfFirstClause/unitTest_funExpressionParasite1/unitTest_funExpressionParasite2/unitTest_funExpressionParasite3/unitTest_funExpressionParasite4/unitTest_bindReceiveFirstClauseParasite/unitTest_bindFunExpressionParasite__IR_OFF/unitTest_bindTry/unitTest_bindFunExpressionFirstClauseParasite/unitTest_bindIf__IR_OFF/unitTest_bindTry__IR_OFF/unitTest_222__IR_OFF/unitTest_newLineAfterArrow1/unitTest_newLineAfterArrow2/unitTest_mapComprehensionAfterExpression/unitTest_bindFunExpressionLastClause/unitTest_bindCaseOf/unitTest_bindBegin/unitTest_bindCaseOfFirstClauseParasite/unitTest_mapTupleAfterLastEntry/unitTest_bindReceiveAfterClause/unitTest_uniformBinaryExpressionsStyle/unitTest_bindReceive/unitTest_bindReceiveParasite__IR_OFF/unitTest_receive1/unitTest_receive2/unitTest_receive3/unitTest_receive4/unitTest_receive5/unitTest_receive6/unitTest_receive7/unitTest_recordFields1/unitTest_recordFields2/unitTest_bindTryExpr__IR_OFF/unitTest_bindReceiveAfter__IR_OFF/unitTest_bindTryLastCatchClauseParasite/unitTest_comprehensions/unitTest_funExpression/unitTest_bindReceiveLastClauseParasite__IR_OFF/unitTest_bindReceiveLastClauseParasite/unitTest_bindReceiveAfter/unitTest_bindFunExpressionFirstClause/unitTest_bindTryExprParasite__IR_OFF/unitTest_binaryComprehensionAfterOrOr/unitTest_bindFunExpressionParasite/unitTest_bindCaseOfLastClauseParasite__IR_OFF/unitTest_recordExpressionEmpty/unitTest_listComprehensionAfterExpression/unitTest_funArgsAfterLast/unitTest_bindCaseOfLastClauseParasite/unitTest_notKeepCommentAtTheFirstLine/unitTest_mapAssocSpacing/unitTest_bindIfLastClauseParasite__IR_OFF/unitTest_bindIfFirstClauseParasite/unitTest_bindCaseParasite/unitTest_functionTypeArguments7
Directory '/tmp/unitTest_afterIf1/unitTest_337/unitTest_noReformatAfterInsertedText/unitTest_functionClause/unitTest_afterIf2/unitTest_ifBeforeIf/unitTest_afterExpression/unitTest_bindTryParasite__IR_OFF/unitTest_bindIfLastClause/unitTest_bindIfFirstClause__IR_OFF/unitTest_bindBeginExpr/unitTest_bindCase__IR_OFF/unitTest_bindBeginExpr__IR_OFF/unitTest_bindReceiveAfterClause__IR_OFF/unitTest_mapTupleBeforeFirstEntry/unitTest_bindCaseOfParasite__IR_OFF/unitTest_bindTryExprParasite/unitTest_bindReceiveAfterParasite/unitTest_bindTryCatch/unitTest_bindIfParasite__IR_OFF/unitTest_bindCaseOfFirstClauseParasite__IR_OFF/unitTest_recordExpressionAfterLastFieldAssignment/unitTest_bindTryCatch__IR_OFF/unitTest_bindTryFirstCatchClause__IR_OFF/unitTest_bindReceiveLastClause/unitTest_listAlignOnEnter/unitTest_binaryComprehensionAfterExpression/unitTest_commaFirstEnter2/unitTest_listComprehensionAfterOrOr/unitTest_funArgsBeforeFirst/unitTest_bindReceiveFirstClause/unitTest_116/unitTest_118/unitTest_125/unitTest_136/unitTest_137/unitTest_141/unitTest_171/unitTest_191/unitTest_222/unitTest_273/unitTest_288/unitTest_292/unitTest_299/unitTest_305/unitTest_350/unitTest_351/unitTest_379/unitTest_433/unitTest_434/unitTest_444/unitTest_451/unitTest_478/unitTest_754/unitTest_if1/unitTest_if2/unitTest_if3/unitTest_funArgsAfterFirst/unitTest_funCallAfterComma/unitTest_aligned/unitTest_case1/unitTest_case2/unitTest_case3/unitTest_case4/unitTest_recordExpressionBeforeFieldAssignment/unitTest_bindBegin__IR_OFF/unitTest_try1/unitTest_try2/unitTest_try3/unitTest_try4/unitTest_try5/unitTest_mapTupleAfterFirstEntry/unitTest_bindBeginParasite__IR_OFF/unitTest_bindCaseExpr__IR_OFF/unitTest_bindReceive__IR_OFF/unitTest_bindReceiveAfterClauseParasite__IR_OFF/unitTest_binaryComprehensionBeforeExpression/unitTest_functionClausesAligned/unitTest_funCallNoArgs/unitTest_binaryComprehensionAfterLastExpr/unitTest_bindIfLastClauseParasite/unitTest_bindTryFirstCatchClauseParasite__IR_OFF/unitTest_caseParasite1/unitTest_caseParasite2/unitTest_caseParasite3/unitTest_caseParasite4/unitTest_bindCaseOfFirstClause/unitTest_recordExpressionAfterFirstFieldAssignment/unitTest_bindCaseOf__IR_OFF/unitTest_48/unitTest_52/unitTest_53/unitTest_67/unitTest_71/unitTest_75/unitTest_82/unitTest_95/unitTest_bindCaseOfLastClause/unitTest_mapComprehensionBeforeExpression/unitTest_bindFunExpressionLastClause__IR_OFF/unitTest_bindBeginExprParasite__IR_OFF/unitTest_bindIfFirstClause/unitTest_funExpressionParasite1/unitTest_funExpressionParasite2/unitTest_funExpressionParasite3/unitTest_funExpressionParasite4/unitTest_bindReceiveFirstClauseParasite/unitTest_bindFunExpressionParasite__IR_OFF/unitTest_bindTry/unitTest_bindFunExpressionFirstClauseParasite/unitTest_bindIf__IR_OFF/unitTest_bindTry__IR_OFF/unitTest_222__IR_OFF/unitTest_newLineAfterArrow1/unitTest_newLineAfterArrow2/unitTest_mapComprehensionAfterExpression/unitTest_bindFunExpressionLastClause/unitTest_bindCaseOf/unitTest_bindBegin/unitTest_bindCaseOfFirstClauseParasite/unitTest_mapTupleAfterLastEntry/unitTest_bindReceiveAfterClause/unitTest_uniformBinaryExpressionsStyle/unitTest_bindReceive/unitTest_bindReceiveParasite__IR_OFF/unitTest_receive1/unitTest_receive2/unitTest_receive3/unitTest_receive4/unitTest_receive5/unitTest_receive6/unitTest_receive7/unitTest_recordFields1/unitTest_recordFields2/unitTest_bindTryExpr__IR_OFF/unitTest_bindReceiveAfter__IR_OFF/unitTest_bindTryLastCatchClauseParasite/unitTest_comprehensions/unitTest_funExpression/unitTest_bindReceiveLastClauseParasite__IR_OFF/unitTest_bindReceiveLastClauseParasite/unitTest_bindReceiveAfter/unitTest_bindFunExpressionFirstClause/unitTest_bindTryExprParasite__IR_OFF/unitTest_binaryComprehensionAfterOrOr/unitTest_bindFunExpressionParasite/unitTest_bindCaseOfLastClauseParasite__IR_OFF/unitTest_recordExpressionEmpty/unitTest_listComprehensionAfterExpression/unitTest_funArgsAfterLast/unitTest_bindCaseOfLastClauseParasite/unitTest_notKeepCommentAtTheFirstLine/unitTest_mapAssocSpacing/unitTest_bindIfLastClauseParasite__IR_OFF/unitTest_bindIfFirstClauseParasite/unitTest_bindCaseParasite' list (0 children): []
[1]: Missing extension point: com.intellij.sdkType in area null
```

**Comment by kvakvs** [comment]
@ignatov is this ready to be delivered?

**Comment by ignatov** [comment]
Please squash all commits together and I'll merge the PR.

**Comment by kvakvs** [comment]
Github squashed it and with squash comes a merge. Sorry 😊

**Pull request merged**